iModelJsNodeAddon/presentation/ECPresentationSerializer.cpp

```diff
@@ -1119,6 +1131,365 @@ rapidjson::Document IModelJsECPresentationSerializer::_AsJson(ContextR ctx, Node
+// @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+DPoint3d GetPoint3dFromJson(RapidJsonValueCR json)
+    {
+    if (json.IsNull() || !json.IsObject())
+        return DPoint3d();
+    return DPoint3d::From(json["x"].GetDouble(), json["y"].GetDouble(), json["z"].GetDouble());
+    }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+ECValue GetECValueFromJson(PrimitiveType type, RapidJsonValueCR json)
+    {
+    ECValue value;
+    if (json.IsNull())
+        {
+        value.SetIsNull(true);
+        return value;
+        }
+
+    switch (type)
+        {
+    case PRIMITIVETYPE_Boolean:
+        value.SetBoolean(json.GetBool());
+        break;
+    case PRIMITIVETYPE_Binary:
+        break;
+    case PRIMITIVETYPE_DateTime:
+        {
+        DateTime dt;
+        if (json.IsDouble())
+            DateTime::FromJulianDay(dt, json.GetDouble(), DateTime::Info::CreateForDateTime(DateTime::Kind::Utc));
+        else
+            DateTime::FromString(dt, json.GetString());
+        value.SetDateTime(dt);
+        break;
+        }
+    case PRIMITIVETYPE_Double:
+        value.SetDouble(json.GetDouble());
+        break;
+    case PRIMITIVETYPE_Integer:
+        value.SetInteger(json.GetInt());
+        break;
+    case PRIMITIVETYPE_Long:
+        if (json.IsString())
+            value.SetLong(BeInt64Id::FromString(json.GetString()).GetValueUnchecked());
+        else
+            value.SetLong(json.GetInt64());
+        break;
+    case PRIMITIVETYPE_String:
+        value.SetUtf8CP(json.GetString());
+        break;
+    case PRIMITIVETYPE_Point2d:
+        value.SetPoint2d(GetPoint2dFromJson(json));
+        break;
+    case PRIMITIVETYPE_Point3d:
+        value.SetPoint3d(GetPoint3dFromJson(json));
+        break;
+    default:
+        DIAGNOSTICS_DEV_LOG(DiagnosticsCategory::Default, LOG_ERROR, Utf8PrintfString("Unrecognized primitive property type: %d", (int)type));
+        }
+    return value;
+    }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+rapidjson::Document IModelJsECPresentationSerializer::_AsJson(ContextR ctx, BoundQueryValuesList const& boundQueryValuesList, rapidjson::Document::AllocatorType* allocator) const
+    {
+    rapidjson::Document json(allocator);
+    json.SetArray();
+    auto boundQueryValueSerializer = IModelJsBoundQueryValueSerializer();
+    for (size_t i = 0; i < boundQueryValuesList.size(); ++i)
+        {
+        auto const& value = boundQueryValuesList.at(i);
+        json.PushBack(value->ToJson(boundQueryValueSerializer, &json.GetAllocator()), json.GetAllocator());
+        }

```

Comment:
see restructure recommendation in `DefaultECPresentationSerializer` for this method

Reply:
Restructured.